### PR TITLE
Add decimal type codec support.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**

### CODE REVIEW

File: util/codec/decimal.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
+// Copyright 2014 The Cockroach Authors.
```
Comment (siddontang) [inline util/codec/decimal.go]: we don't need add cockroach license here.

File: util/codec/decimal.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+// EncodeInt    -> value sign
+// EncodeInt    -> exp sign
+// EncodeInt    -> exp value
+// EncodeBytes  -> abs value bytes
+func EncodeDecimal(b []byte, d mysql.Decimal) []byte {
+	if d.Equals(mysql.ZeroDecimal) {
+		return EncodeInt(b, zeroSign)
+	}
+
+	valSign := codecSign(int64(d.BigIntValue().Sign()))
```
Comment (siddontang) [inline util/codec/decimal.go]: var v := d.BigIntValue() first

File: util/codec/decimal.go (hunk 1 above)
Comment (qiuyesuifeng) [inline util/codec/decimal.go]: got it

File: util/codec/decimal.go (hunk 2 above)
Comment (qiuyesuifeng) [inline util/codec/decimal.go]: ok

File: util/codec/decimal_test.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
+// Copyright 2014 The Cockroach Authors.
```
Comment (siddontang) [inline util/codec/decimal_test.go]: here too

Comment (qiuyesuifeng) [inline util/codec/decimal_test.go]: yes, my fault.

File: util/codec/codec_test.go [hunk 4]
```diff
@@ -513,3 +513,82 @@ func (s *testCodecSuite) TestDuration(c *C) {
...
+		{"-1.234", "-12.34", 1},
+		{"-0.1234", "-12.34", 1},
+		{"-12.34", "1234", -1},
+		{"-12.34", "-12.35", 1},
+		{"-0.01234", "-0.01235", 1},
+		{"-1234", "-123400", 1},
+		{"-12340", "-123400", 1},
+	}
+
+	for _, t := range tblCmp {
```
Comment (siddontang) [inline util/codec/codec_test.go]: add some boundary value test? e.g, maxint8, maxint8 +- 1?

**Comment by siddontang** [comment]
No desc codec?, e.g, EncodeDecimalDesc

**Comment by siddontang** [comment]
ignore decimal desc, we can use EncodeBytesDesc directly.

**Comment by coocood** [comment]
LGTM

**Pull request merged**